### ISSUE TITLE
vue@3 support

### ISSUE THREAD

**Issue by kirillgroshkov** [vuematerial/vue-material#2277]
Would be nice to support Vue 3 :)

**Comment by mikimoresi** [comment]
Once Vue 3 goes out of beta and first official realese will be well tested it should pretty straight forward to migrate vue material to Vue 3.

**Comment by kirillgroshkov** [comment]
> Once Vue 3 goes out of beta and first official realese will be well tested it should pretty straight forward to migrate vue material to Vue 3.

It already did: https://github.com/vuejs/vue-next/releases/tag/v3.0.0

**Comment by mikimoresi** [comment]
> > Once Vue 3 goes out of beta and first official realese will be well tested it should pretty straight forward to migrate vue material to Vue 3.
> 
> It already did (https://github.com/vuejs/vue-next/releases/tag/v3.0.0), hence this issue:)

too early in my opinion, vue core team is also building a tool to help migrate from 2 to 3.
Of course if vue material core team decide to migrate sooner i will be happy to :)

**Comment by DevVue** [comment]
Would be great to have this, we are looking to convert a Vue 2 app to Vue 3, but stuck here as have used Vue material for the project. Any news on when this will be available?

**Comment by prgmr6** [comment]
we too are looking to get in on Vue 3, please let us know when to expect this

**Comment by sinisarudan** [comment]
Yes, we're also stuck on Vue 2 due to our dependency on Vue Material :)
So would be very glad to hear news on **vue@3 support** Thanks in advance and thanks for your work so far!

**Comment by narenderv7** [comment]
Is it supporting Vue 3?

**Comment by Suyashtnt** [comment]
nope

**Comment by Maxhodges** [comment]
Is this even being worked on? is there a roadmap? ETA? 
Cheers!

**Comment by allanvobraun** [comment]
Any workaround?

**Comment by mikimoresi** [comment]
> Any workaround?

fork repo and convert it for vue 3 :)

**Comment by allanvobraun** [comment]
> > Any workaround?
> 
> fork repo and convert it for vue 3 :)

If I do that, will you merge the project?

**Comment by mikimoresi** [comment]
I'm not an owner, if you di that i will for sure use your fork :)

**Comment by dragosct10** [comment]
Hi guys! If some of you will do the vue3 update we will be happy to merge the PR. 

Regards,
Dragos

**Comment by allanvobraun** [comment]
I wish I had the time to do that :(

**Comment by mikimoresi** [comment]
me neither, is not a simple operation.
Anyway I think rushing to Vue 3 is not so important. It has great new features but Vue 2 is still solid and mature.
Time to time.

**Comment by NotoriousRebel** [comment]
Any updates on this issue? Was going to use this for a project and now can't because it doesn't support vue3 🙃

**Comment by kogratte** [comment]
Same here. I've a POC to develop, and I really don't want to be stucked with vue@2 and its undecent support of TS..

**Comment by Usouf** [comment]
What are the updates here?

**Comment by vartdalen** [comment]
Any updates?

**Comment by lyssar** [comment]
Seems that this is a kinda dead horse by now? Feels like it.

**Comment by baradhili** [comment]
vue team's migrate tool as well as codemods.. I prefer vue v3 and don't want to re-invent the wheel :)

**Comment by mgriffoul** [comment]
Still no updates ?

**Comment by RonaldTayie** [comment]
This might take some time, so I'd suggest sticking to Vue2 until 3 is stable enough to transition to. I'm still new to VueJS and Vue3 hasn't got much to offer in term of compatibility with older modules. So I'm sticking to 2 for now.

**Comment by artkad** [comment]
Still no updates ?

**Comment by tyt2y3** [comment]
We are about to purchase Vue Material Kit PRO, would be nice to know the roadmap going forwards :)

**Comment by taearls** [comment]
it would be great to get an idea of the roadmap for vue 3 support. my team is also stuck in vue 2 to support this library. 

if there's an open branch to address this maybe some additional stakeholders can contribute.

**Comment by marqbeniamin** [comment]
Hi everyone, I created a new branch for vue 3 support: https://github.com/vuematerial/vue-material/tree/dev-vue3.
If you want to help migration to vue 3, you can do the pull request in this branch. Thank you for your support!

**Comment by mamzellejuu** [comment]
@marqbeniamin do you have a release date for the migration?

**Comment by marqbeniamin** [comment]
Hi @mamzellejuu , thank you for your interest in our project!
We are in talks with a team of developers who want to help us. When we have all the details I come back with an update here.

All the best,
Beniamin

**Comment by CodyEddings** [comment]
Are there any updates?

**Comment by sam-berry** [comment]
Willing to help on this if help is needed.

**Comment by RobQuistNL** [comment]
![image](https://user-images.githubusercontent.com/1442796/155364071-aadf5d3a-8a8e-42b3-acc1-ebdc95f2f855.png)

**Comment by alexvenetis** [comment]
> ![image](https://user-images.githubusercontent.com/1442796/155364071-aadf5d3a-8a8e-42b3-acc1-ebdc95f2f855.png)

I tried converting my vue-material / vue 2 project to vue 3 and couldn't get it running at all. Looks like it will require a complete re-write to get it working...

**Comment by rafael-chaves-ic** [comment]
Any news on the migration to vue3?

**Comment by mikimoresi** [comment]
the repo is clearly abandoned or at least very bad maintained..
I have my forks docs working better than the officials....

I will go on using as little as possible vue material components so that i will be able to convert only that ones only and when Vue3 will be really necessary...

**Comment by gsimon75** [comment]
Thanks y'all for this thread 🙏, I _almost_ started experimenting with vue-material. This has spared me quite some struggle to reach the same conclusion.

**Comment by goldyfruit** [comment]
> gsimon75

Same here.

**Comment by goldyfruit** [comment]
> Thanks y'all for this thread pray, I _almost_ started experimenting with vue-material. This has spared me quite some struggle to reach the same conclusion.

Same here.

**Comment by zdravkov** [comment]
Pity about this library :( - one possible alternative that has great Material theme and supports Vue 3 is [Kendo UI for Vue](https://www.telerik.com/kendo-vue-ui).

**Comment by goldyfruit** [comment]
> Pity about this library :( - one possible alternative that has great Material theme and supports Vue 3 is [Kendo UI for Vue](https://www.telerik.com/kendo-vue-ui).

This is not a free alternative.

I was looking as well at [Vuetify](https://github.com/vuetifyjs/vuetify) but same, only beta version available for Vue 3 with a lot of missing `props`.

I decided to go with [Quasar](https://github.com/quasarframework/quasar), supports Vue 3 and works with TypeScript.

**Comment by zdravkov** [comment]
Yes indeed :) . Well it depends what you are looking for.

**Comment by pedersenderekserif** [comment]
Pretty please? I loved using this lib, but the vue 3 migration makes this impossible. Does the paid for version play nice with vue3?

**Comment by RobQuistNL** [comment]
Nope, using paid version here.

**Comment by mikimoresi** [comment]
I am converting my project to vue3, this project was using some ( 10/15 components) of vue material..so I toke this components and embedded them directly in my project to convert them to vue3 while going on developing and converting also my other components, they are now part of my project, not a library anymore.
I am close to finish this conversion, then I will provide a repo with the converted components, it will not be in the form of a library but you will be able to take them and embed in your projects like I am doing or even reconvert it to library if you want.

**Comment by mikimoresi** [comment]
as promised, here is the repo:
[https://github.com/mikimoresi/vue3-material](https://github.com/mikimoresi/vue3-material)

In this repo you can find several vue material components converted to vue3, in the `app.vue` you can see how to use them. Actually you need to directly embed them in your code but if you have the time a reconversion to library should be very easy.

I converted all the components that I was using but tabs and tables, replacing them with alternatives looking very similar was much faster and easyer than converting them.

I am using [Varlet](https://varlet.gitee.io/varlet-ui/#/en-US/tabs) for tabs and [vue3-table-lite](https://github.com/linmasahiro/vue3-table-lite) for tables. 

Generally speaking this conversion has been not too easy, unfortunately vue-material library looks very good but the code is far to be optimal, there are several particular techniques used all across the library that was not to be considered closed to standards even with vue2.

If someone wants to partecipate to the development and optimization of vue3-material is of course very welcome.

Cheers

MM

**Comment by akinoccc** [comment]
Hey guys, I have been developing the web components of [material vue3](https://github.com/akinocccc/material-v3) and base on [material design v3](https://m3.material.io/). but now just only myself, if someone has interest in it, welcome pr.
